tests/js/dataMock/analytics.js

```diff
@@ -0,0 +1,311 @@
+			case 'TIME':
+				values.push( faker.random.number( { min: 0, max: 3600 } ).toString() ); // 1 hour max.
+				break;
+			case 'CURRENCY':
+				values.push( faker.random.float( { min: 0, max: 10000 } ).toString() ); // $10k max.
+				break;
+		}
+
+		metrics.push( { values } );
+	}
+
+	return metrics;
+}
+
+/**
+ * Sorts report rows and returns it.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array.<Object>}        rows    Array of rows to sort.
+ * @param {Array.<Object>}        metrics Array of report metrics.
+ * @param {Object|Array.<Object>} orderby Sorting options.
+ * @return {Array.<Object>} Sorted rows.
+ */
+function sortRows( rows, metrics, orderby ) {
+	let sorted = rows;
+
+	const orders = Array.isArray( orderby ) ? orderby : [ orderby ];
+	for ( const order of orders ) {
+		const direction = order?.sortOrder === 'DESCENDING' ? -1 : 1;
+		const index = metrics.findIndex( ( metric ) => getMetricKey( metric ) === order?.fieldName );
+		if ( index < 0 ) {
+			continue;
+		}
+
+		sorted = sorted.sort( ( a, b ) => {
+			let valA = parseFloat( a.metrics[ index ]?.values?.[ 0 ] );
+			if ( Number.isNaN( valA ) ) {
+				valA = 0;
+			}
+
+			let valB = parseFloat( b.metrics[ index ]?.values?.[ 0 ] );
+			if ( Number.isNaN( valB ) ) {
+				valB = 0;
+			}
+
+			return ( valA - valB ) * direction;
+		} );
+	}
+
+	return sorted;
+}
+
+/**
+ * Generates mock data for Analytics reports.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} args Report options.
+ * @return {Array.<Object>} An array with generated report.
+ */
+export function getAnalyticsMockResponse( args ) {
+	// We set seed for every data mock to make sure that the same arguments get the same report data.
+	// It means that everyone will have the same report data and will see the same widgets in the storybook.
+	// This approach gives us additional flexebility to control randomness on per widget basis.
+	faker.seed( getObjectHash( args ) );
+
+	const data = {
+		dataLastRefreshed: null,
+		isDataGolden: null,
+		rowCount: 0,
+		samplesReadCounts: null,
+		samplingSpaceSizes: null,
+		rows: [],
+		totals: [],
+		minimums: [],
+		maximums: [],
+	};
+
+	const { compareStartDate, compareEndDate } = args;
+	const metricValuesCount = compareStartDate && compareEndDate ? 2 : 1;
+
+	const validMetrics = ( args.metrics || [] ).filter( ( metric ) => !! getMetricType( metric ) );
+	const streams = [];
+
+	// Generate streams (array) of dimension values. Each dimension will have its own stream (array) of data.
+	// Then streams will be merged into one (see zip( ... ) function call) and metric values will be added to each
+	// dimension set in the combined stream (array). We need to use array of streams because report arguments may
+	// have 0 or N dimensions (N > 1) which means that in the each row of the report data we will have an array
+	// of dimension values.
+	const dimensions = Array.isArray( args.dimensions ) ? args.dimensions : [ args.dimensions ];
+	dimensions.forEach( ( dimension ) => {
```

Comment:
There might be a case when the `dimensions` option isn't provided at all. Or it can be a single string value, or an array of dimensions. We need to account for different cases.

Reply:
As mentioned before, this should be possible to normalize so that we wouldn't have to account for all variants in all places.